workflow/controller/operator.go

```diff
@@ -1975,11 +2029,12 @@ func getStepOrDAGTaskName(nodeName string) string {
 
 func (woc *wfOperationCtx) executeScript(nodeName string, templateScope string, tmpl *wfv1.Template, orgTmpl wfv1.TemplateReferenceHolder, opts *executeTemplateOpts) (*wfv1.NodeStatus, error) {
 	node := woc.getNodeByName(nodeName)
-	if node != nil {
+	if node != nil && !(tmpl.Synchronization != nil && node.Pending()) {
```

Comment:
This logic seems overly complex and it's not easy to understand what's going on. Could you explain what the intention is here and simplify the logic?

Reply:
It seems the reason is because `if node != nil` needs to be sprinkled in everywhere. I'm not sure why `executeScript` is necessary to be modified, but I would argue that it should not be necessary to be modified.